hail/src/test/scala/is/hail/expr/ir/EmitStreamSuite.scala

```diff
@@ -304,4 +310,32 @@ class EmitStreamSuite extends HailSuite {
         IndexedSeq(null, 0L, 2L, 7L, 15L, 15L, 15L, 16L, 17L)
     )
   }
+
+  @Test def testEmitFromIterator() {
+    val iter = Ref("iter", EmitStreamSuite.iterType)
```

Comment:
Could you add a test where the iterator is an argument to the compiled function? Also, we should test the cases where the iterator is missing, and where some of its values are missing.

Reply:
done, but required refactoring how im testing the streams in EmitStreamSuite a bit

Reply:
emitstream case for `Ref` is gone too replaced with just `In`. the multiplicity check is part of the map partitions PR #7681 where its actually important